l10n_it_account/models/account_tax.py

```diff
@@ -47,21 +39,64 @@ def _compute_deductible_balance(self):
     )
     def _compute_undeductible_balance(self):
         for tax in self:
-            account_ids = (
-                tax.mapped("invoice_repartition_line_ids.account_id")
-                | tax.mapped("refund_repartition_line_ids.account_id")
-            ).ids
-            balance_regular = tax.compute_balance(
-                tax_or_base="tax",
-                financial_type="regular",
-                exclude_account_ids=account_ids,
+            account_ids = tax._get_accounts_tax().ids
+            tax.undeductible_balance = tax._compute_tax_balance_by_accounts(
+                exclude_account_ids=account_ids
             )
-            balance_refund = tax.compute_balance(
-                tax_or_base="tax",
-                financial_type="refund",
-                exclude_account_ids=account_ids,
+
+    @api.depends_context(
+        "from_date",
+        "to_date",
+        "company_ids",
+        "target_move",
+    )
+    def _compute_debit_balance(self):
+        for tax in self:
+            accounts = tax._get_accounts_tax()
+            accounts = accounts.filtered(
+                lambda a: a.account_type.startswith("liability")
```

Comment:
Qui e nell'altro `startswith`, penso sia meglio indicare esplicitamente quali sono i tipi di conto da considerare come debito/credito, come viene fatto ad esempio per i tipi di registrazioni contabili in https://github.com/odoo/odoo/blob/44e61c6b429aab94e4656ade5e0f0275b905210a/addons/account/models/account_move.py#L5524-L5533.

Reply:
modificato